IdentityCore/src/cache/metadata/MSIDMetadataCache.m

```diff
@@ -138,11 +158,13 @@ - (BOOL)removeAccountMetadataForKey:(MSIDCacheKey *)key
     __block NSError *localError;
     
     dispatch_sync(_synchronizationQueue, ^{
-        [_memoryCache removeObjectForKey:key];
-        
         success = [_dataSource removeAccountMetadataForKey:key context:context error:&localError];
```

Comment:
should this line also be in the following `dispatch_barrier_async`? otherwise some corner case of race condition may happen :)

Reply:
You're right, will update. Thanks